src/main/java/frc/robot/subsystems/LedSubsystem.java

```diff
@@ -0,0 +1,124 @@
+package frc.robot.subsystems;
+
+
+import edu.wpi.first.wpilibj.AddressableLED;
+import edu.wpi.first.wpilibj.AddressableLEDBuffer;
+
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+
+// our robot constants
+
+import frc.robot.Constants.LedConstants;
+
+
+
+public class LedSubsystem extends SubsystemBase {
+
+/**
+ * The LED subsystem provides all of the methods for driving the LEDs on the robot
+ */
+
+    private AddressableLED m_led;
+
+    private AddressableLEDBuffer m_ledBuffer;
+
+
+
+    public LedSubsystem() {
+
+        m_led = new AddressableLED(LedConstants.kLedPwmPort);
+
+        m_ledBuffer = new AddressableLEDBuffer(LedConstants.kNumOfLeds);
+        m_led.setLength(LedConstants.kNumOfLeds);
+
+        // setLedsMaroon();
+   
+        m_led.start();
+    }
+    
+
+
+    public int getNumOfLeds() {
+        return m_ledBuffer.getLength();
+    }
+
+
+
+    // all the various color and effect routines do not ned to do a .setData() call
+    //
+    // the periodic routine for the LedSubsystem will do it for you
+
+    public void setLedsOff() {
+
+        for (var i = 0 ; i < m_ledBuffer.getLength() ; i++) {
```

Comment:
I notice that for these following for loops, the initial variable is of type var. Is there an advantage for using var over i in these methods?

Reply:
Not really.  It's a common pattern you'll see to basically assign a type based on the assignment (an integer in this case).

For us, it might actually be clearer to just make it an int declaration since that's what it turns out to be.